Extensions/postarchive.js

```diff
@@ -75,6 +88,16 @@ XKit.extensions.postarchive = {
 
 		this.running = true;
 
+		window.addEventListener("message", function(e) {
+			if (e.origin !== "https://www.tumblr.com" && !e.data.xkit) { return; }
+			if (e.data.extension_method) {
+				XKit.extensions.postarchive[e.data.extension_method].apply(null, e.data.args);
+			}
+			if (e.data.expand_in_blog_control_iframe) {
+				$('.tmblr-iframe--desktop-loggedin-controls').width('100%');
+			}
+		});
```

Comment:
wait i think there's a misunderstanding as to what expand_in_blog_control_iframe does.  it doesn't replace the old XKit.iframe.resize stuff, it just makes the iframe as wide as necessary to hold the extra buttons.